### ISSUE TITLE
preload() hangs on loadImage

### ISSUE THREAD

**Issue by slowizzm** [processing/p5.js-web-editor#674]
preload() hangs when trying to loadImage(), never loads.

Google Chrome | 68.0.3440.106 (Official Build) (64-bit) (cohort: Stable)
Revision | 1c32c539ce0065a41cb79da7bfcd2c71af1afe62-refs/branch-heads/3440@{#794}
OS | Windows

**Comment by catarak** [comment]
i'm not able to reproduce this! the follow sketch works for me: https://editor.p5js.org/cassie/sketches/SJlIk0WUPX

can you add a link to your sketch? thanks!

**Comment by slowizzm** [comment]
Weird, it is working for me now as well, your sketch and mine.

No longer an issue.

thanks

**Comment by mberndtgen** [comment]
Sorry for re-opening this, but I was able to reproduce this. In preload(), I'm loading 15 images into an array like so:
```javascript
  for (var i = 0; i < numOfImages; i++) {
    imgs[i] = loadImage("images/" + i + "_image.png");
  }
```
and draw wasn't able to show them. Only after modifying my code and 'forcing' them to display (like so:
```javascript
function draw() {
  image(imgs[0], 0, 0);
}
```
- doing this for each picture, one by one, made me eventually display them as intended. After this procedure my original code worked as expected. Felt a bit like filling a cache. Definitely weird and maybe worth a look.

**Comment by mberndtgen** [comment]
It's here: https://editor.p5js.org/mberndtgen/sketches/B12RmDyvX

**Comment by datramt** [comment]
I got the sketch to work by replacing all of the "var" declarations to "let"

**Comment by mberndtgen** [comment]
Strange, huh? :-) This code worked without problems in the alpha version of the editor.

**Comment by catarak** [comment]
@mberndtgen i was playing with your sketch and i couldn't figure out what was wrong because of all of the other stuff going on in that project (which is really cool btw!). would you mind making an example that only shows the thing that appears to not be working?

**Comment by mberndtgen** [comment]
Yeah, sure! (And thanks! This is my homework for a course on Kadenze.) Here is the essence of what it takes (at least for me) to reproduce the error:
https://editor.p5js.org/mberndtgen/sketches/ByBKxP2vQ

It was even a bit stranger than expected: I've cloned this project from the original one (via File > Duplicate) and modified it. I've changed the png's in the images folder, too (but they still have the names of the original project: 0_image.png, 1_image.png etc.) When I run this code then imgs[0] is displayed as an image of my original project, while the same image that was loaded directly (not into an array) is displayed correctly. See here: 

![strange-p5editor-behavior](https://user-images.githubusercontent.com/7542281/45055128-450b9900-b08f-11e8-8674-62f1f3c5bbc4.png)

Background: image from array, green thingy: correct image. Both were loaded as 0_image.png. This is different from my first experience when no image of the array was displayed at all, but I guess this could have the same cause.

**Comment by slowizzm** [comment]
preload() loadImage() array is working for me ....

https://editor.p5js.org/slow_izzm/sketches/BJnX6QJ_X

**Comment by catarak** [comment]
@mberndtgen this is what I see:
<img width="631" alt="screen shot 2018-09-07 at 12 51 57 pm" src="https://user-images.githubusercontent.com/6063380/45232287-e156bb00-b29c-11e8-92e4-9525a9427c9f.png">
so I think for me it's... working? I'm feeling very confused about what exactly the issue you're seeing is. I am also using Chrome.

**Comment by mberndtgen** [comment]
Oh weirdness! Now it's working on my side, too! Can't reproduce the error any longer. You better close this issue until this changes again. :-) Thanks a lot for your effort, though, I really appreciate!
(Adventure's almost finished, btw.)

**Comment by catarak** [comment]
maybe it's a caching thing, or a network connection thing. i'll close this for now!

**Comment by slowizzm** [comment]
This may need to be re-opened. 

I have no issues with preload() in safari or firefox, but on Chrome 70 on macOS (High Sierra) preloads hangs on loadImage and never loads.

https://editor.p5js.org/slow_izzm/sketches/r1OcQ45om

**Comment by catarak** [comment]
@slowizzm thanks! i'll try updating chrome and see if i can reproduce.

**Comment by slowizzm** [comment]
> @slowizzm thanks! i'll try updating chrome and see if i can reproduce.

thank you ... looks like you fixed it, it's working as expected now.

**Comment by catarak** [comment]
i think there's sometimes an issue loading assets from S3, which comes up unpredictably.

**Comment by catarak** [comment]
This issue comes up intermittently but I've never been able to pinpoint the exactly cause. Sometimes S3 will just send back a CORS issue, and then moments later it won't. Not sure! I'm going to close this but feel free to open a new issue if it comes up again.

**Comment by selazbin-s** [comment]
first link posted (https://editor.p5js.org/cassie/sketches/SJlIk0WUPX) is not working

**Comment by eyaler** [comment]
i am seeing loadImage() failing specifically in preload() on macOS. same sketch works on windows. sketch works on macOS if loadImage is moved into setup(). should i open a new issue or could this be reopened?

**Comment by raclim** [comment]
Thanks for reporting this @eyaler! We can reopen this issue!
